executor/builder.go

```diff
@@ -692,10 +693,20 @@ func (b *executorBuilder) buildProjection(v *plan.PhysicalProjection) Executor {
 }
 
 func (b *executorBuilder) buildTableDual(v *plan.PhysicalTableDual) Executor {
-	return &TableDualExec{
-		baseExecutor: newBaseExecutor(v.Schema(), b.ctx),
-		rowCount:     v.RowCount,
+	if v.RowCount != 0 && v.RowCount != 1 {
+		b.err = errors.Errorf("buildTableDual failed, invalid row count for dual table: %v", v.RowCount)
+		return nil
+	}
+	dualSchema := v.Schema()
+	if dualSchema.Len() == 0 {
```

Comment:
Why change the schema?

Reply:
the schema can not be empty, we need to use it to build a chunk for `TableDualExec` and append some value into it to indicate there exists some rows in that chunk.

Reply:
this work should be done in plan building stage